source/common/grpc/async_client_impl.h

```diff
@@ -49,9 +49,10 @@ class AsyncClientImpl final : public AsyncClient<RequestType, ResponseType> {
 
   AsyncStream<RequestType>* start(const Protobuf::MethodDescriptor& service_method,
                                   AsyncStreamCallbacks<ResponseType>& callbacks) override {
+    const Optional<std::chrono::milliseconds> no_timeout;
```

Comment:
Why was this change needed to fix ASAN? This still feels kind of busted to me. Aren't we storing a reference to the timeout inside the AsyncStreamImpl? If so should this be at client scope?